cachito/workers/cleanup_job.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+import logging
+from datetime import datetime, timedelta
+
+import requests
+
+from cachito.workers.requests import get_requests_session
+from cachito.workers.config import get_worker_config
+from cachito.errors import CachitoError
+
+log = logging.getLogger(__name__)
+
+auth_session = get_requests_session(auth=True)
+session = get_requests_session()
+config = get_worker_config()
+
+
+def main():
+    """
+    Get all completed requests from the Cachito API.
+
+    :raise CachitoError: if the request to the Cachito API fails
+    """
+    # Get all completed requests
+    response = session.get(config.cachito_api_url + 'requests?state=complete')
+    if not response.ok:
+        raise CachitoError('Could not reach Cachito API to get all completed requests')
+    json_response = response.json()
+    identify_and_mark_stale_requests(json_response)
+
+
+def identify_and_mark_stale_requests(requests_json):
+    """
+    Identify completed Cachito requests which have reached end of life and mark them as stale.
+
+    :param dict requests_json: json representation of all complete requests
+    """
+    stale_request_ids = []
+    current_time = datetime.utcnow()
+    for request in requests_json['items']:
+        date_time_obj = datetime.strptime(request['updated'], '%Y-%m-%dT%H:%M:%S.%f')
+        if current_time - date_time_obj > timedelta(config.cachito_request_lifetime):
+            stale_request_ids.append(request['id'])
```

Comment:
Optional: It'd be more efficient if you marked the requests as stale as you identify them instead of finding them all and then marking them. Basically O(n) instead of O(2n).

Reply:
But the complexity in this case remains the same which is O(kn) == O(n). Doing it separately just makes it more readable. :)

Reply:
@yashvardhannanavati how so? You're currently looping through the requests to find the ones to mark and then you loop through those requests to mark them as stale. My suggestion is as you loop through the requests the first time, just mark them as stale as you identify them. That way you loop through the requests once, not twice.

Reply:
Changed it but this is what I meant [Why is O(kn) == O(n)](https://stackoverflow.com/questions/13161397/is-okn-linear-complexity-or-quadratic-complexity-or-does-it-depend-on-the-k)

In the grand scheme of things, unless k is O(n) (nested for loops), a constant value of k doesn't increase the complexity of an algorithm. Ofcourse, O(2n) will run a bit slower than O(n) but that is a very minimal difference. :)